File: olnatura-qr/web/qr-enterprise-frontend/src/utils/labelDocumentCode.ts
export const LABEL_DOCUMENT_CODE = "AL-001-E02/06";
const LEGACY_DOCUMENT_CODE = "AL-001-E02/04";

export function resolveLabelDocumentCode(code?: string | null): string {
  const raw = String(code ?? "").trim();
  if (!raw) return LABEL_DOCUMENT_CODE;
  if (raw.startsWith(LEGACY_DOCUMENT_CODE)) {
    return LABEL_DOCUMENT_CODE + raw.slice(LEGACY_DOCUMENT_CODE.length);
  }
  return raw;
}

File: olnatura-qr/api/olnaturaqr/src/main/java/com/company/olnaturaqr/support/label/LabelDocumentCode.java
package com.company.olnaturaqr.support.label;

public final class LabelDocumentCode {

    public static final String CURRENT = "AL-001-E02/06";
    private static final String LEGACY = "AL-001-E02/04";

    private LabelDocumentCode() {}

    public static String resolve(String stored) {
        String raw = stored == null ? "" : stored.trim();
        if (raw.isEmpty()) return CURRENT;
        if (raw.startsWith(LEGACY)) {
            return CURRENT + raw.substring(LEGACY.length());
        }
        return raw;
    }
}
